### PR TITLE
DetailsList: Introduce new styles via DisplayMode prop

### PR DESCRIPTION
**Pull request by leddie24**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

- Adds in DisplayMode prop to DetailsList, which is a enum prop for either the original, new, or compact mode styles.  
- Sets default prop to Original 36px row height styles
- Resets Check component to original size for Original DetailsList variant

Screenshot comparison:

**Original (no selection)**:
original-no-selection

**Original (with selection)**:
original-selection

**New Style (no selection)**:
new-no-selection

**New Style (with selection)**:
new-selection

**Compact mode (no selection)**:
compact-no-selection

**Compact mode (with selection)**:
compact-selection

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
@leddie24 can you take a look at these conflicts?

**Comment by micahgodbolt** [comment]
@leddie24 also, can you drop in a few screenshots of the updates you did

**Comment by leddie24** [comment]
@micahgodbolt fixed conflicts and updated with screenshots.

**Comment by leddie24** [comment]
@ThomasMichon  can you take a look when you get a chance please?  Thanks!

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts [hunk 1]
```diff
@@ -400,4 +403,13 @@ export enum CheckboxVisibility {
    * Hide checkboxes.
    */
   hidden = 2
+}
+
+export enum DetailsListDisplayMode {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.Props.ts]: Can't we use theming for that?

File: packages/office-ui-fabric-react/src/components/Check/Check.scss [hunk 2]
```diff
@@ -70,4 +71,28 @@ $checkBoxHeight: 16px;
     border-radius: 10px;
     border: 1px solid WindowText;
   }
+}
+
+:global(.ms-DetailsList--Original) {
+  .root {
+    width: $origCheckBoxHeight - 2px;
```
Comment (battletoilet) [inline packages/office-ui-fabric-react/src/components/Check/Check.scss]: What is the 2px value representing here? Maybe we could make it a sass variable.

File: packages/office-ui-fabric-react/src/components/Check/Check.scss [hunk 3]
```diff
@@ -70,4 +71,28 @@ $checkBoxHeight: 16px;
...
+    height: $origCheckBoxHeight - 2px;
```
Comment (battletoilet) [inline packages/office-ui-fabric-react/src/components/Check/Check.scss]: Same here

**Comment by dzearing** [comment]
Closing this PR as it hasn't had enough traction. If we need this change, please revive it.

**Pull request closed without merging**